include/babeltrace/compat/socket-internal.h

```diff
@@ -0,0 +1,395 @@
+#ifndef _BABELTRACE_COMPAT_SOCKET_H
+#define _BABELTRACE_COMPAT_SOCKET_H
+
+/*
+ * babeltrace/compat/socket.h
+ *
+ * Copyright (C) 2015-2017  Michael Jeanson <mjeanson@efficios.com>
+ *               2015  Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifdef __MINGW32__
+
+#include <winsock2.h>
+
+#define BT_INVALID_SOCKET INVALID_SOCKET
```

Comment:
Agreed for `BT_INVALID_SOCKET` not being an error code, that's a good point.

As for the naming scheme, I would just unify everything under an internal `bt_socket` library to group all of these utilities. In the future, I would move the rest of the socket calls (recv() and others) there too to keep everything consistent.

The rationale behind this is that our current compatibility scheme of having _loose_ `bt_*`-prefixed functions makes it hard to figure where a compatibility wrapper must be used and where something can be used "as-is".

In the end, I typically merge a patch, the build fails on some platform/config, and then I have to figure whether or not a wrapper already exists.

Since sockets seem to require a fair amount of platform-specific code, I think keeping an internal socket library will make everything simpler in the long run.